src/java.base/unix/classes/sun/nio/fs/UnixNativeDispatcher.java

```diff
@@ -544,6 +545,13 @@ static boolean openatSupported() {
         return (capabilities & SUPPORTS_OPENAT) != 0;
     }
 
+    /**
+     * Supports EISDIR error of unlink(2)
+     */
+    static boolean eisdirSupported() {
+        return (capabilities & SUPPORTS_EISDIR) != 0;
```

Comment:
The approach in this version looks much better. Need to find a better name for the capability and method name as "EISDIR" is supported on all Unix/like platforms. I think we want something to convey unlink detects directories or fails for directories.

Reply:
Naming updated in [3814a18](https://github.com/openjdk/jdk/pull/25107/commits/3814a184551496b01d9f0f44a514c23aa1e094b7).

Reply:
Updated name looks okay, thanks for adjusting that.